optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/locator/LocationStrategyResolver.java

```diff
@@ -88,6 +89,21 @@ public LocationStrategy determineLocationStrategy(Object object) {
                     }
                     return new PlanningIdLocationStrategy(memberAccessor2);
                 case EQUALITY:
+                    try {
+                        Method equals = object.getClass().getMethod("equals", Object.class);
+                        if (equals.getDeclaringClass().equals(Object.class)) {
+                            throw new IllegalArgumentException("The class (" + object.getClass().getSimpleName()
+                                    + ") doesn't override equals method, neither does any superclass.");
+                        }
+                        Method hashCode = object.getClass().getMethod("hashCode");
+                        if (hashCode.getDeclaringClass().equals(Object.class)) {
+                            throw new IllegalArgumentException("The class (" + object.getClass().getSimpleName()
+                                    + ") overrides equals but neither it nor any superclass overrides hashCode method."
+                            );
+                        }
```

Comment:
Do we also make sure that `hashCode` and `equals` are overridden in the same class? In that case, fail or only log a warning? For example, class overriding equals and relying on hashCode from superclass is weird but it could work because even if the subclass's equals is more specific, equal objects still would have the same hashCode. The other way around (hashCode in subclass) would likely result in equal objects with different hashCode, which breaks hashCode's contract.